odbc/src/api/get_info_tests.rs

```diff
@@ -281,102 +296,221 @@ mod unit {
+        expected_value = MONGO_CA2_SUPPORT,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_dynamic_cursor_attributes1,
+        info_type = InfoType::SQL_DYNAMIC_CURSOR_ATTRIBUTES1 as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        expected_length = std::mem::size_of::<u32>() as i16,
+        expected_value = SQL_CA1_NEXT,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_dynamic_cursor_attributes2,
+        info_type = InfoType::SQL_DYNAMIC_CURSOR_ATTRIBUTES2 as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        expected_length = std::mem::size_of::<u32>() as i16,
+        expected_value = MONGO_CA2_SUPPORT,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_static_cursor_attributes1,
+        info_type = InfoType::SQL_STATIC_CURSOR_ATTRIBUTES1 as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        expected_length = std::mem::size_of::<u32>() as i16,
+        expected_value = SQL_CA1_NEXT,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_static_cursor_attributes2,
+        info_type = InfoType::SQL_STATIC_CURSOR_ATTRIBUTES2 as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        expected_length = std::mem::size_of::<u32>() as i16,
+        expected_value = MONGO_CA2_SUPPORT,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_scroll_options,
+        info_type = InfoType::SQL_SCROLL_OPTIONS as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        expected_length = std::mem::size_of::<u32>() as i16,
+        expected_value = MONGO_SO_SUPPORT,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_bookmark_persistence,
+        info_type = InfoType::SQL_BOOKMARK_PERSISTENCE as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        expected_length = std::mem::size_of::<u32>() as i16,
+        expected_value = 0,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_need_long_data_len,
+        info_type = InfoType::SQL_CATALOG_NAME as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        buffer_length = 2 * size_of::<WideChar>() as i16,
+        expected_length = size_of::<WideChar>() as i16,
+        expected_value = "Y",
+        actual_value_modifier = modify_string_value,
+    );
+
+    test_get_info!(
+        sql_txn_isolation_option,
+        info_type = InfoType::SQL_TXN_ISOLATION_OPTION as u16,
+        expected_sql_return = SqlReturn::SUCCESS,
+        expected_length = std::mem::size_of::<u32>() as i16,
+        expected_value = SQL_TXN_SERIALIZABLE,
+        actual_value_modifier = modify_u32_value,
+    );
+
+    test_get_info!(
+        sql_database_name,
+        info_type = InfoType::SQL_DATABASE_NAME as u16,
```

Comment:
It seems difficult to test the "current catalog is some" case for this info type via these tests, curious if it came up at all in your POC work. Could be useful to add (possibly in an integration test) sometime in the future!

Reply:
We do have a test for the ConnectAttr version, but you're correct we don't have it for this SQLGetInfo version. I can probably just add an integration test now.

Reply:
Also, now I see that we are returning NO_DATA, not test, for the ConnectAttr version, and we should probably be consistent

Reply:
Ok, added a test where we set the catalog first.

Reply:
Why would it matter? I never see any issue with supporting extra behavior.

Reply:
What matters is that adding "extra behaviour" is hard to track. It is also not expected by tools/users.
Not supporting extra behaviour means that you don't have to think twice or more every time you are deviating from the spec to make sure that it won't break anything.
In this case for example, I don't see any harm. But I am not the end user and if they rely on the call to fail if the connection was not made successfully, then they would be right, even if it seems very silly and they probably should rely on something else like the status of the their SQLxxConnect call.